tests/test_openclhpp.cpp

```diff
@@ -3833,4 +3833,46 @@ void testSemaphoreRetain(void) {}
 void testSemaphoreRelease(void) {}
 #endif // cl_khr_semaphore
 
+/*
+ * Tests for allocate and deallocate API (clSVMAlloc, clSVMFree, clEnqueueSVMMap)
+ ****************************************************************************/
+static int testMemory[3];
+
+void *clSVMAlloc_stubForMemoryAllocation(cl_context context,cl_svm_mem_flags flags,
+                         size_t size, cl_uint alignment, int cmock_num_calls)
+{
+  
+    TEST_ASSERT_BITS(CL_MEM_READ_WRITE, flags, CL_MEM_READ_WRITE);
+    TEST_ASSERT_EQUAL_HEX(size, 3 * sizeof(int));
+    TEST_ASSERT_EQUAL_HEX(0, alignment);
+    return testMemory;
+}
+cl_int clEnqueueSVMMap_stubForMemoryAllocation(cl_command_queue command_queue, cl_bool blocking_map, cl_map_flags flags, void* svm_ptr, size_t size, cl_uint num_events_in_wait_list, const cl_event* event_wait_list, cl_event* event, int cmock_num_calls)
+{
+    TEST_ASSERT_EQUAL_PTR(svm_ptr, testMemory);
+    TEST_ASSERT_EQUAL_HEX(size, 3 * sizeof(int));
+    return CL_SUCCESS;
+}
+void clSVMFree_stubForMemoryAllocation(cl_context context, void *svm_pointer,
+    int cmock_num_calls)
+{
+    TEST_ASSERT_EQUAL_PTR(svm_pointer, testMemory);
+}
+void testSVMMemoryAllocation()
+{
+
+#if CL_HPP_TARGET_OPENCL_VERSION >= 200
+    cl::SVMAllocator<int, cl::SVMTraitCoarse<>> svmAllocator;
+        
+    clSVMAlloc_StubWithCallback(clSVMAlloc_stubForMemoryAllocation);
+    clEnqueueSVMMap_StubWithCallback(clEnqueueSVMMap_stubForMemoryAllocation);
+    clSVMFree_StubWithCallback(clSVMFree_stubForMemoryAllocation);
+
+    int *ptr = svmAllocator.allocate(3);
+    TEST_ASSERT_EQUAL_PTR(ptr,testMemory);
+    clSVMFree_Expect(0, ptr);
```

Comment:
I can't figure out what this line is doing - is it needed?

Reply:
Sorry about that, this is a case where the GitHub UI didn't do me any favors.

I'm asking about this line specifically: `clSVMFree_Expect(0, ptr);`

We already have `clSVMFree_StubWithCallback` so is the additional `clSVMFree_Expect` needed?  The test seems to work fine when I remove it, and I can't figure out why `clSVMFree` would be called twice.